### PR TITLE
Adding support for multiple mask tokens.

### PR DESCRIPTION
**Pull request by Narsil**
# What does this PR do?

- Original implem: https://github.com/huggingface/transformers/pull/10222

When presented multiple masks, it's impossible to retrieve the conjugate probabilities.
Instead of trying to workaround that (see discussions in previous PR) this PR instead
just outputs the raw `top_k` proposition at each locus, since it gets trickier to find a good
proxy for "joint probabilities". Instead of trying to solve this impossible problem we simply
show exactly what the model outputs.

@naveenjafer mentionned as co-author since much of this PR was pulled from there.

This PR was resurrected partly because Perceiver (byte-level model) need to do this type of masking to be useful.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by LysandreJik — COMMENTED** [review]
This looks good to me, cool work @Narsil. 

Regarding the conversations in https://github.com/huggingface/transformers/pull/10222, should this be marked as experimental for the first few weeks so that conversation may occur on this feature?

Could you add a note about the support of this feature in the documentation?

File: src/transformers/pipelines/fill_mask.py [hunk 1]
```diff
@@ -59,13 +59,13 @@ def get_masked_index(self, input_ids: GenericTensor) -> np.ndarray:
...
-                self.model.base_model_prefix,
-                f"More than one mask_token ({self.tokenizer.mask_token}) is not supported",
-            )
-        elif numel < 1:
+        # if numel > 1:
+        #     raise PipelineException(
+        #         "fill-mask",
+        #         self.model.base_model_prefix,
+        #         f"More than one mask_token ({self.tokenizer.mask_token}) is not supported",
+        #     )
```
[the remark this answers is not in the document]
Comment (Narsil) [reply src/transformers/pipelines/fill_mask.py]: Yes, thanks.

File: tests/test_pipelines_fill_mask.py [hunk 2]
```diff
@@ -340,3 +338,27 @@ def fill_mask_with_duplicate_targets_and_top_k(self, model, tokenizer):
         # The target list contains duplicates, so we can't output more
         # than them
         self.assertEqual(len(outputs), 3)
+
+    def fill_mask_with_multiple_masks(self, model, tokenizer):
```
Comment (NielsRogge) [inline tests/test_pipelines_fill_mask.py:342]: Can we perhaps add a test for Perceiver (similar to the image classification models)?

File: src/transformers/pipelines/__init__.py [hunk 3]
```diff
@@ -224,33 +236,48 @@
...
 
+NO_FEATURE_EXTRACTOR_TASKS = set()
+NO_TOKENIZER_TASKS = set()
+for task, values in SUPPORTED_TASKS.items():
+    if values["type"] == "text":
+        NO_FEATURE_EXTRACTOR_TASKS.add(task)
+    elif values["type"] in {"audio", "image"}:
+        NO_TOKENIZER_TASKS.add(task)
+    elif values["type"] != "multimodal":
+        raise ValueError(f"SUPPORTED_TASK {task} contains invalid type {values['type']}")
```
[the remark this answers is not in the document]
Comment (NielsRogge) [reply src/transformers/pipelines/__init__.py:279]: Looks good to me as well!

File: tests/test_pipelines_fill_mask.py (hunk 2 above)
Comment (Narsil) [reply tests/test_pipelines_fill_mask.py:342]: I think this PR is pretty orthogonal to Perceiver.

We could add a slow test for sure, but it doesn't have to be Perceiver specific.

In fact, I'll add something on the random model (it just needs to be consistent, actual values are less important)

**Review by LysandreJik — APPROVED** [review]
Ok this looks good to me! Looking forward to the additional test, feel free to merge whenever.

**Pull request merged**